### PR TITLE
Showing gauge install instruction on welcome page when gauge is not installed

### PR DESCRIPTION
**Pull request by Apoorva-GA**
When gauge binaries are not found but gauge-vscode plugin is installed, welcome page shows a section where gauge install instructions are shown based on the OS.

### CODE REVIEW

**Comment by sguptatw** [comment]
Is there a way to bring up the Gauge welcome page?
> I can bring up VSCode Help Welcome page from the command pallet

File: src/constants.ts [hunk 1]
```diff
@@ -31,7 +31,8 @@ export enum GaugeVSCodeCommands {
     RepeatExecution = 'gauge.execute.repeat',
     SwitchProject = 'gauge.specexplorer.switchProject',
     QuickPickOnExecution = 'gauge.quickPick.onExecution.open',
-    ExtractConcept = 'gauge.extract.concept'
+    ExtractConcept = 'gauge.extract.concept',
+    CreateAndSendTextToInstallTerminal = "gauge.createAndSendText.terminal"
```
Comment (sriv) [inline src/constants.ts]: `CreateAndSendTextToInstallTerminal ` sounds like a very loaded name. Can we have something simpler, ex `ExecuteInTerminal`?

Also consider simplifying `gauge.createAndSendText.terminal`

File: src/extension.ts [hunk 2]
```diff
@@ -46,13 +46,22 @@ let treeDataProvider: Disposable = new Disposable(() => undefined);
...
 let outputChannel: OutputChannel = window.createOutputChannel('gauge');
 let specExplorerActiveFolder: string = "";
+let terminalStack: Terminal[] = [];
 
 export function activate(context: ExtensionContext) {
+    let currentExtensionVersion = extensions.getExtension(GAUGE_EXTENSION_ID)!.packageJSON.version;
+    let hasUpgraded = hasExtensionUpdated(context, currentExtensionVersion);
+    context.subscriptions.push(new WelcomePageProvider(context, hasUpgraded));
+
+    context.subscriptions.push(commands.registerCommand
```
Comment (sriv) [inline src/extension.ts]: Consider moving the command + registration to a separate class. Even `terminalStack` can be encapsulated in the new class. See `WelcomeProvider` for example.

File: src/ui/welcome/index.html [hunk 3]
```diff
@@ -9,12 +9,14 @@
...
 			--inversion: 1; /* logo is white background, light themes need an inversion */
 			--header-colour: #555555;
+			--command-background : rgba(255,255,255,0.08);
 		}
 		.vscode-dark{
 			--button-background: rgba(255,255,255,0.08);
 			--button-hover-background: rgba(255,255,255,0.1);
 			--inversion: 0;
 			--header-colour: #e0b009;
+			--command-background : black;
```
Comment (sriv) [inline src/ui/welcome/index.html]: Why is there an opacity in light theme, but none in dark theme for `--command-background`?

File: src/ui/welcome/index.html [hunk 4]
```diff
@@ -132,6 +134,59 @@
...
+			background-color:	var(--command-background);
+			color:white;
+			padding: 10px 20px 10px 20px;
+		}
+		.instructions {
+			justify-content: left;
+			align-items: center;
+			border-radius: 10px;
+			padding: 10px 20px 10px 20px;
+			/* background-color: #9a363e; */
```
Comment (sriv) [inline src/ui/welcome/index.html]: remove commented styles

File: src/welcome/welcome.ts [hunk 5]
```diff
@@ -35,13 +37,49 @@ export class WelcomePageProvider extends Disposable implements TextDocumentConte
...
+    }
+
     async provideTextDocumentContent(uri: Uri): Promise<string> {
         let rootPath = path.join('out', uri.path);
         let docPath = Uri.file(this._context.asAbsolutePath(path.join(rootPath, 'index.html')));
         const doc = await workspace.openTextDocument(docPath);
         let text =  doc.getText();
         let supress = this._context.globalState.get<Boolean>(GAUGE_SUPPRESS_WELCOME);
         return text
+            .replace(/{{installCommand}}/g, encodeURI('command:gauge.createAndSendText.terminal?' +
```
Comment (sriv) [inline src/welcome/welcome.ts]: There are too many chained replace. It now makes sense to extract a method and unit test it.

File: src/welcome/welcome.ts [hunk 6]
```diff
@@ -35,13 +37,49 @@ export class WelcomePageProvider extends Disposable implements TextDocumentConte
...
+                installCommand.name = "Chocolatey";
+                installCommand.command = "choco install gauge";
+                return installCommand;
+            case "darwin":
+                installCommand.name = "Brew";
+                installCommand.command = "brew install gauge";
+                return installCommand;
+            default:
+                if (this.getLinuxDistribution().indexOf("Ubuntu") !== -1) {
+                    installCommand.name = "apt-get";
```
Comment (sriv) [inline src/welcome/welcome.ts]: debian package manager is `apt`. 

Also lets be consistent on the casing. `Brew` and `Chocolatey` have caps in the name, `apt` and `dnf` do not.

Consider making them `brew`, `choco`, `apt` and `dnf`

File: src/welcome/welcome.ts [hunk 7]
```diff
@@ -35,13 +37,49 @@ export class WelcomePageProvider extends Disposable implements TextDocumentConte
...
+                    installCommand.command = "sudo apt-get install gauge";
```
Comment (sriv) [inline src/welcome/welcome.ts]: If the command is running in `sudo`, this can be highlighted in the install section. Perhaps as a footnote.

**Pull request merged**